openwisp_radius/tests/test_api.py

```diff
@@ -2386,8 +2427,77 @@ def test_change_phone_number_restriction(self):
 
         app_settings.ALLOWED_MOBILE_PREFIXES = []
+
+    def _test_change_phone_number_sms_on_helper(self, is_active):
+        self.test_register_201()
+        user = User.objects.get(email=self._test_email)
+        user_token = Token.objects.filter(user=user).last()
+        phone_token_qs = PhoneToken.objects.filter(user=user)
+        url = reverse('radius:phone_number_change', args=[self.default_org.slug])
+        old_phone_number = '+237675582041'
+        user.phone_number = old_phone_number
+        user.is_active = is_active
+        user.save()
+        new_phone_number = '+237675582042'
+
+        self.assertEqual(phone_token_qs.count(), 0)
+
+        r = self.client.post(
+            url,
+            json.dumps({'phone_number': new_phone_number}),
+            content_type='application/json',
+            HTTP_AUTHORIZATION=f'Bearer {user_token.key}',
+        )
+        self.assertEqual(r.status_code, 200)
+        self.assertEqual(phone_token_qs.count(), 1)
+        self.assertEqual(phone_token_qs.first().phone_number, new_phone_number)
+        user.refresh_from_db()
+        self.assertEqual(user.phone_number, old_phone_number)
+        self.assertFalse(user.is_active)
+
+    def test_active_user_change_phone_number_sms_on(self):
+        self._test_change_phone_number_sms_on_helper(True)
+
+    def test_inactive_user_change_phone_number_sms_on(self):
+        self._test_change_phone_number_sms_on_helper(False)
+
+    def _test_change_phone_sms_off_helper(self, is_active):
+        self.test_register_201()
+        user = User.objects.get(email=self._test_email)
+        user_token = Token.objects.filter(user=user).last()
+        phone_token_qs = PhoneToken.objects.filter(user=user)
+        radius_settings = self.default_org.radius_settings
+        radius_settings.sms_verification = False
+        radius_settings.full_clean()
+        radius_settings.save()
+        url = reverse('radius:phone_number_change', args=[self.default_org.slug])
+        old_phone_number = '+237675582041'
+        user.phone_number = old_phone_number
+        user.is_active = is_active
+        user.save()
+        new_phone_number = '+237675582042'
+
+        self.assertEqual(phone_token_qs.count(), 0)
+
+        r = self.client.post(
+            url,
+            json.dumps({'phone_number': new_phone_number}),
+            content_type='application/json',
+            HTTP_AUTHORIZATION=f'Bearer {user_token.key}',
+        )
+        self.assertEqual(r.status_code, 200)
+        self.assertEqual(phone_token_qs.count(), 0)
+        user.refresh_from_db()
+        self.assertEqual(user.phone_number, new_phone_number)
+        self.assertEqual(user.is_active, is_active)
+
+    def test_active_user_change_number_sms_off(self):
+        self._test_change_phone_sms_off_helper(True)
+
+    def test_inactive_user_change_number_sms_off(self):
+        self._test_change_phone_sms_off_helper(False)
```

Comment:
please simplify the tests so they assume sms verification is always on.

When it's off the usage of the endpoint will be rejected by #189.